crates/trie/parallel/src/root.rs

```diff
@@ -261,6 +277,27 @@ impl From<alloy_rlp::Error> for ParallelStateRootError {
     }
 }
 
+/// Gets or creates a tokio runtime handle for spawning blocking tasks.
+/// This ensures we always have a runtime available for I/O operations.
+fn get_runtime_handle() -> Handle {
```

Comment:
hmm, this will most likely create a new runtime with a bunch of threads because the tree task is run outside of tokio

so we'll endup with a bunch of unusable threads. which should be fine since we dont expect this to be reachable in the happy path.

and this is better than sprinkling the runtime as arg all over the place